### PR TITLE
Added environment variables support

### PR DESCRIPTION
**Pull request by matthiasantierens**
Added Jaeger environment variables.

### CODE REVIEW

File: molecule/default/playbook.yml [hunk 1]
```diff
@@ -3,3 +3,8 @@
   any_errors_fatal: true
   roles:
     - name: ansible-cortex
+  vars:
+    cortex_env_variables:
```
Comment (roidelapluie) [inline molecule/default/playbook.yml]: the default scenario should have no values

Comment (roidelapluie) [reply molecule/default/playbook.yml]: also, where are the values tested?

File: templates/cortex.service.j2 [hunk 2]
```diff
@@ -10,6 +10,7 @@ StartLimitIntervalSec=0
 Type=simple
 User={{ cortex_system_user }}
 Group={{ cortex_system_group }}
+EnvironmentFile=-{{ cortex_environment_location }}/cortex
```
Comment (roidelapluie) [inline templates/cortex.service.j2:13]: /cortex is a variable too

File: templates/cortex@.service.j2 [hunk 3]
```diff
@@ -9,6 +9,7 @@ StartLimitIntervalSec=0
 [Service]
 Type=simple
 User={{ cortex_system_user }}
+EnvironmentFile=-{{ cortex_environment_location }}/cortex-%i
```
Comment (roidelapluie) [inline templates/cortex@.service.j2:12]: /cortex is a variable too

File: tasks/configure.yml [hunk 4]
```diff
@@ -32,3 +32,10 @@
   notify:
     - reload cortex
   loop: "{{ cortex_services | dict2items }}"
+
+- name: Configure cortex environment file
+  template:
+    src: "cortex.environment.j2"
+    dest: "{{ cortex_environment_location }}/{{ cortex_environment_file }}"
```
Comment (roidelapluie) [inline tasks/configure.yml]: where is the cortex_all_in_one version?

**Review by roidelapluie — CHANGES_REQUESTED** [review]

File: molecule/services/tests/test_services.py [hunk 5]
```diff
@@ -47,7 +48,7 @@ def test_service(host):
     assert s.is_running
 
 
-@pytest.mark.parametrize("port", [9009, 9010, 9095, 9096])
+@pytest.mark.parametrize("port", [9009, 9010, 9096])
```
Comment (roidelapluie) [inline molecule/services/tests/test_services.py]: why is that one gone?

File: templates/cortex@.environment.j2 [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{{ ansible_managed | comment }}
+{# LOOP over environment file to make jaeger envonment file #}
+
+{% for name,value in cortex_env_variables.items() %}
```
Comment (roidelapluie) [inline templates/cortex@.environment.j2]: That does bot seem right, we should only have the variables for the current service

File: molecule/services/tests/test_services.py (hunk 5 above)
Comment (matthiasantierens) [reply molecule/services/tests/test_services.py]: This port is not being used in the playbook.

File: molecule/alternative/tests/test_alternative.py [hunk 7]
```diff
@@ -29,6 +29,7 @@ def test_directories(host, dirs):
     "/cortex/cortex.yml",
     "/etc/systemd/system/cortex.service",
     "/usr/local/bin/cortex-linux-amd64",
+    "/etc/default/cortex",
```
Comment (roidelapluie) [inline molecule/alternative/tests/test_alternative.py:32]: can you test the content of the file?

File: molecule/services/tests/test_services.py [hunk 8]
```diff
@@ -29,6 +29,7 @@ def test_directories(host, dirs):
     "/cortex/cortex-ingester.yml",
     "/etc/systemd/system/cortex@.service",
     "/usr/local/bin/cortex-linux-amd64",
+    "/etc/default/cortex-ingester",
```
Comment (roidelapluie) [inline molecule/services/tests/test_services.py:32]: can you test the content of the file?

File: molecule/services/tests/test_services.py [hunk 9]
```diff
@@ -63,3 +64,13 @@ def test_config_file_explicit_target(host):
...
+
+
+def test_string(host):
+    f = host.file("/etc/default/cortex-ingester")
+    assert "KAEGER_AGENT_HOST=localhost" in f.content_string
+
+
+def test_string2(host):
+    f = host.file("/etc/default/cortex-readpath")
+    assert "KAEGER_SAMPLER_TYPE=probabilistic" in f.content_string
```
Comment (roidelapluie) [inline molecule/services/tests/test_services.py]:
```suggestion
    assert "KAEGER_AGENT_HOST: localhost\n" in f.content_string
    assert "KAEGER_SAMPLER_TYPE=probabilistic\n" in f.content_string
```

**Comment by roidelapluie** [comment]
Thanks!

**Comment by roidelapluie** [comment]
Can you update README ?

**Pull request merged**